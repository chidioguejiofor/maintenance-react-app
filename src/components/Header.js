import React from 'react';

const Header = () => (
  <div id="header">
    <img
      src="https://image.flaticon.com/icons/svg/949/949890.svg"
      alt="site logo"
    />
    <span>
        Maintenance Tracker
    </span>
  </div>

);

export default Header;
