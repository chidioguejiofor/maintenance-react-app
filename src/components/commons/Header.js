import React, { Component } from 'react';


/**
 * This class contains code for the Header component
 */
class Header extends Component {
  /**
   * this renders content to the DOM
   *@returns {JSX} returns content to be rendered in the DOM
   */
  render() {
    return (
      <h1>Header  is Header</h1>
    );
  }
}

export default Header;
