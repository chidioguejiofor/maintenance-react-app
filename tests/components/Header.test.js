/* eslint no-undef:off */
import React from 'react';
import { mount, shallow } from 'enzyme';
import Header from '../../src/components/Header';

describe('Testing the Header Component', () => {
  it('should contain an image of some sort', (done) => {
    const wrapper = mount(<Header />);
    const div = wrapper.find('div');

    expect(div.length).toBeGreaterThan(0);
    expect(div.children().length).toEqual(2);
    expect(div.first().find('img').length).toEqual(1);
    done();
  });
  it('should contain an span with text Maintenance Tracker', (done) => {
    const wrapper = mount(<Header />);
    const div = wrapper.find('div');
    expect(div.find('span').first().text())
      .toEqual('Maintenance Tracker');

    done();
  });

  it('should have an image link that is equal to what is passed as props', () => {
    const src = 'tempSrc';
    const wrapper = shallow(<Header imgSrc={src} />);
    expect(wrapper).toMatchSnapshot();
  });
});
