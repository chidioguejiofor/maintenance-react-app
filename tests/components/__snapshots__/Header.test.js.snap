// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Header Component should have an image link that is equal to what is passed as props 1`] = `
<div
  className="header"
>
  <img
    alt="site logo"
    src="https://image.flaticon.com/icons/svg/949/949890.svg"
  />
  <span>
    Maintenance Tracker
  </span>
</div>
`;
